pnpm/src/pnpm.ts

```diff
@@ -14,14 +14,34 @@ const argv = process.argv.slice(2)
     console.log(version)
     break
   }
+  case 'completion': {
+    const { getCompletionScript } = await import('@pnpm/tabtab')
+    function exitError (message: string): never {
+      console.error(message)
+      process.exit(1)
+    }
+    const shell = argv[1]?.trim()
+    if (!shell) {
+      exitError('missing argument for shell')
+    }
+    if (!['bash', 'fish', 'pwsh', 'zsh'].includes(shell)) {
+      exitError(`${shell} is not supported`)
+    }
+    const completionScript = await getCompletionScript({ name: 'pnpm', completer: 'pnpm', shell })
+    console.log(completionScript)
+    return
+  }
   case 'install-completion': {
     const { install: installCompletion } = await import('@pnpm/tabtab')
     await installCompletion({ name: 'pnpm', completer: 'pnpm', shell: argv[1] })
     return
   }
   case 'uninstall-completion': {
     const { uninstall: uninstallCompletion } = await import('@pnpm/tabtab')
-    await uninstallCompletion({ name: 'pnpm' })
+    await Promise.all(
+      ['bash', 'fish', 'pwsh', 'zsh']
+        .map((shell) => uninstallCompletion({ name: 'pnpm', shell }))
+    )
     return
   }
```

Comment:
let's refactor this and create a new completion plugin. So that it works the same as other commands.